File: rincon_del_cafe/models/Book.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace rincon_del_cafe.models
{
    [Table("libros")]
    public class Book
    {
        [Key]
        [Column("id")]
        public int Id { get; set; }

        [Column("isbn")]
        public string? Isbn { get; set; }

        [Column("titulo")]
        public string Titulo { get; set; }

        [Column("autor")]
        public string Autor { get; set; }

        [Column("editorial")]
        public string Editorial { get; set; }

        [Column("imagen_portada")]
        public string ImagenPortada { get; set; }

        [Column("publishing_year")]
        public int PublishingYear { get; set; }

        [Column("paginas")]
        public int Paginas { get; set; }

        [Column("genero")]
        public string Genero{ get; set; }

        [Column("descripcion")]
        public string Descripcion { get; set; }
    }
}

File: book.js
const apiUrl = "https://localhost:7092/api/book"
const container = document.getElementById("books-container")

fetch(apiUrl)
    .then(data => {
        if (!data.ok) {
            throw new Error("Network error")
        }        

        return data.json();
    })
    .then(res => {
        console.log(res);

        res.forEach(el => {
            const card = `                
                <div class="book-card" id="book-card">
                <h2>${el.titulo}. Publicado: ${el.publishingYear}</h2>
                <p>${el.autor}</p>
                <p>${el.genero}</p>
                <img src="${el.imagenPortada}">
                <p>${el.descripcion}</p>
                <p>${el.editorial}</p>
                <p>páginas: ${el.paginas}</p>
                <p>ISBN: ${el.isbn}</p>
                </div>
            `
        container.insertAdjacentHTML("beforeend", card)
        });
        
    }) 
    

    .catch(err => {
        console.error("Error fetching the data", err);

        container.innerHTML = `
         <p style="color:red;">Something went wrong, try again later </p> 
        `
    })